test/boost/sstable_set_test.cc

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (C) 2020 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <seastar/core/sstring.hh>
+#include <seastar/core/future-util.hh>
+#include <seastar/core/align.hh>
+#include <seastar/core/aligned_buffer.hh>
+#include "sstables/compaction.hh"
+#include <seastar/testing/test_case.hh>
+#include <seastar/testing/thread_test_case.hh>
+#include "test/boost/sstable_test.hh"
+#include <seastar/core/seastar.hh>
+#include <seastar/core/do_with.hh>
+#include "sstables/compaction_strategy_impl.hh"
+#include "sstables/date_tiered_compaction_strategy.hh"
+#include "sstables/time_window_compaction_strategy.hh"
+
+using namespace sstables;
+
+static const sstring some_keyspace("ks");
+static const sstring some_column_family("cf");
+
+static shared_sstable sstable_for_overlapping_test(test_env& env, const schema_ptr& schema, int64_t gen, sstring first_key, sstring last_key, uint32_t level = 0) {
+    auto sst = env.make_sstable(schema, "", gen, la, big);
+    sstables::test(sst).set_values_for_leveled_strategy(0, level, 0, std::move(first_key), std::move(last_key));
+    return sst;
+}
+
+using sstable_set_opt = std::optional<sstable_set>;
+using sstable_list_opt = std::optional<sstable_list>;
+
+SEASTAR_TEST_CASE(test_versioned_sstable_set_test) {
```

Comment:
This is almost a palindrome.

Reply:
I'm simply using the full type in the next versions